src/GraphQL/Execution/ExecutionHelper.cs

```diff
@@ -367,9 +367,14 @@ public static ArgumentValue CoerceValue(ISchema schema, IGraphType type, IValue
 
                 if (input is ListValue list)
                 {
-                    return new ArgumentValue(list.Values
-                        .Select(item => CoerceValue(schema, listItemType, item, variables).Value)
-                        .ToList(), ArgumentSource.Literal);
+                    var count = list.ValuesList.Count;
+                    if (count == 0)
+                        return new ArgumentValue(Array.Empty<object>(), ArgumentSource.Literal);
+
+                    var values = new object[count];
+                    for (int i = 0; i < count; ++i)
+                        values[i] = CoerceValue(schema, listItemType, list.ValuesList[i], variables).Value;
+                    return new ArgumentValue(values, ArgumentSource.Literal);
```

Comment:
This now returns an array instead of a `List<object>`.  I think that's fine...

Reply:
ok

Reply:
I had written it so that `ParseDictionary` only needs to process nested sets of `Dictionary<string, object>` or `List<object>`.